components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIDefinition.java

```diff
@@ -266,4 +268,40 @@ public abstract String processDisableSecurityExtension(String swaggerContent)
      * @return String parserType
      */
     public abstract String getType();
+
+    /**
+     * This method will extract the root level swagger extension
+     *
+     * @param swaggerContent String
+     * @param extensionKey   String
+     * @return Object
+     * @throws APIManagementException
+     */
+    public abstract Object getRootLevelSwaggerExtension(String swaggerContent, String extensionKey);
+
+    /**
+     * This method will extract the resource level swagger extension
+     *
+     * @param swaggerContent String
+     * @return {@link ThrottlingLimit}
+     * @throws APIManagementException if the value under extension cannot be mapped to a {@link ThrottlingLimit}
+     *                                object
+     */
+    public ThrottlingLimit getAPILevelThrottlingLimit(String swaggerContent)
+            throws APIManagementException {
+        Object extension = getRootLevelSwaggerExtension(swaggerContent,
+                APIConstants.X_THROTTLING_LIMIT_OPENAPI_EXTENSION);
+        ObjectMapper mapper = new ObjectMapper();
+        try {
+            if (extension != null) {
+                return mapper.convertValue(extension, ThrottlingLimit.class);
+            }
+            // Unreachable condition as the extension is already validated before calling this method.
+        } catch (IllegalArgumentException e) {
+            throw new APIManagementException(
+                    String.format("Invalid content under the %s extension in swagger definition",
+                            APIConstants.X_THROTTLING_LIMIT_OPENAPI_EXTENSION), e);
```

Comment:
Better not to stick to OpenAPI or Swagger here as this is getting added to AsyncAPI Parser too